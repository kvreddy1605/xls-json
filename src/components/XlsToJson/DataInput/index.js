import React from 'react';
import './index.scss';

class DataInput extends React.Component {
	constructor(props){
		super(props);
		this.handleChange = this.handleChange.bind(this);
	}

	handleChange(e){
		const files = e.target.files;
		if(files && files[0]) this.props.handleFile(files[0]);
	};

	render() { 
		return (
			<div className="dataInput-container" >
				<form>
					<label>Browse a file with type xls or csv.</label> <br />
					<input type="file" id="file" accept=".xls, .xlsx, .csv, .xlsb, .xlsm" onChange={this.handleChange} />
				</form>
			</div>
		); 
  	}
}

export default DataInput;