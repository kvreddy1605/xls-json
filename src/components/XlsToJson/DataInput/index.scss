.dataInput-container{
    input{
        margin: 20px;
    }
}