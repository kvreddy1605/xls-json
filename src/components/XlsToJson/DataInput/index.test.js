import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DataInput from './index';

describe('DataInput Component Test', () => {
    it('SnapShot Test', () => {
        const wrapper = shallow(<DataInput />)
    
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('change Event trigger, handleFile is called', () => {
        const handleFile = jest.fn();
        const wrapper = shallow(<DataInput handleFile={handleFile}/>)
        wrapper.find('input').simulate('change', { target: {files: [1,2,3]} });
        expect(handleFile).toHaveBeenCalledWith(1);
    })
})
