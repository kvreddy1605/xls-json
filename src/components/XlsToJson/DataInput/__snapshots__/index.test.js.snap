// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataInput Component Test SnapShot Test 1`] = `
<div
  className="dataInput-container"
>
  <form>
    <label>
      Browse a file with type xls or csv.
    </label>
     
    <br />
    <input
      accept=".xls, .xlsx, .csv, .xlsb, .xlsm"
      id="file"
      onChange={[Function]}
      type="file"
    />
  </form>
</div>
`;
